src/c-writer.cc

```diff
@@ -2614,6 +2596,43 @@ bool CWriter::IsImport(const std::string& name) const {
   return import_module_sym_map_.count(name);
 }
 
+template <typename sources>
+void CWriter::Spill(const TypeVector& types, bool ptr, const sources& src) {
+  for (Index i = 0; i < types.size(); ++i) {
+    Write("tmp", ptr ? "->" : ".");
+    Writef("%c%d = ", MangleType(types.at(i)), i);
+    Write(src(i), ";", Newline());
```

Comment:
So these always spill and unspill to a struct `tmp` that is assumed that have the correct members.

The mangling of the type seem unnecessary but maybe makes things easier to debug/read?   i.e  just is the index `i` should be enough to make each field unique?

Reply:
Hmm, good point. This is copied from the current code for multi-value returns and tags, but I think you're right that we could avoid the mangled types and call the fields just "val0" "val1" "val2" or something. (We need **some** alpha prefix since I don't think we can name a field literally just "0" or "1".) Anyway I guess let's keep this as-is for now and we can strip it down in a future change if anybody wants.